stable/pomerium/templates/tls-secrets.yaml

```diff
@@ -16,3 +16,129 @@ data:
 ---
 {{- end }}
 {{- end }}
+
+{{- define "pomerium.authenticate.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
```

Comment:
@travisgroth shouldn't this `generateTLS` option simply disappear with this new chart?

Reply:
The intent is to retain the ability to auto generate the certs for users to keep the adoption bar low.  For users there's two improvements depending on how you handle SSL:

For users that are fine with generated cert chains:
- Shift into into tls secrets that helm can "abandon" after the hook runs.  It is a little wonky as it takes two runs to get settled but then helm diff will stop seeing differences and helm will stop regenerating certs.  If you're leveraging a tool like helmfile this is really important.

For users sourcing certs from elsewhere: 
- Shift into discrete tls secrets that other tools more easily integrate with.  This should improve usability whether you're originating from cert manager, manual kubectl creation, or injecting at helm install time.

Reply:
@travisgroth so, as a user that wants to have auto-generated certs, I will set `generateTLS` to `true` and leave it to `true` at all time, not just the first time, right?

Reply:
Yes.  You can leave it on.  IIRC `true`->`false` should also work IIRC but I'd need to double check that doesn't have unintended consequences.

Reply:
ok, I think I got it:
- if I let this chart generate certificates, I keep `generateTLS` to true, they will be generated the first time and then ignored by helm/helmfile.
- if I generate the certificate myself, then I don't set `generateTLS` but I still need this chart to generate the secret with the value I provided via values.
- if I generate the certificate myself and provide the secrets myself, how do I do that?

Reply:
>if I let this chart generate certificates, I keep generateTLS to true, they will be generated the first time and then ignored by helm/helmfile.

👍 

>if I generate the certificate myself, then I don't set generateTLS but I still need this chart to generate the secret with the value I provided via values.

set `generateTLS` false, and provide via values

>if I generate the certificate myself and provide the secrets myself, how do I do that?

Set `authenticate.existingTLSSecret`, `authorize.existingTLSSecret`, and `proxy.existingTLSSecret`.  I can add a blurb about this in the new section about certificates.

Reply:
> Set authenticate.existingTLSSecret, authorize.existingTLSSecret, and proxy.existingTLSSecret. I can add a blurb about this in the new section about certificates.

Ok, I had trouble with the various if/else and didn't see that there were checks for those value to be set in the `not generateTLS` branch. So it's seems it's good, thank you for your patience :)

Reply:
No problem at all.  Thanks for the feedback.